plugins/modules/proxmox.py

```diff
@@ -415,11 +424,25 @@ def is_template_container(self, node, vmid):
         return config['template']
 
     def create_instance(self, vmid, node, disk, storage, cpus, memory, swap, timeout, clone, **kwargs):
+
+        # Version limited features
+        minimum_version = {
+            'tags': 7,
+        }
         proxmox_node = self.proxmox_api.nodes(node)
 
         # Remove all empty kwarg entries
         kwargs = dict((k, v) for k, v in kwargs.items() if v is not None)
 
+        version = self.version()
+        pve_major_version = 3 if version < LooseVersion('4.0') else version.version[0]
+
+        # Fail on unsupported features
+        for option, version in minimum_version.items():
+            if pve_major_version < version:
+                self.module.fail_json(changed=False, msg="Feature {option} is only supported in PVE {version}+, and you're using PVE {pve_major_version}".
```

Comment:
You need to check whether the option is actually in use:
```suggestion
                if option in kwargs:
                    self.module.fail_json(changed=False, msg="Feature {option} is only supported in PVE {version}+, and you're using PVE {pve_major_version}".
```
(The next line needs to be indented as well.)

Reply:
I feel a bit stupid - this should be fixed :)